homeassistant/components/withings/__init__.py

```diff
@@ -36,14 +54,22 @@
 )
 
 
-async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Withings component."""
     conf = config.get(DOMAIN, {})
     if not conf:
         return True
 
+    # Make the config available to the oauth2 config flow.
     hass.data[DOMAIN] = {const.CONFIG: conf}
 
+    # Initialize a webhook handler and api view.
+    init_webhook_processor(hass)
+    webhook.async_register(
+        hass, const.DOMAIN, "Withings notify", const.WEBHOOK_ID, async_webhook_handler
```

Comment:
We should not use a fixed webhook ID. Instead generate one with the `async_generate_id` function in webhook and store it in the config entry.

Reply:
Example https://github.com/home-assistant/core/blob/dev/homeassistant/components/netatmo/__init__.py#L108-L146